src/main/java/org/ibp/api/java/impl/middleware/study/StudyServiceImpl.java

```diff
@@ -431,4 +359,142 @@ public Integer getEnvironmentDatasetId(final Integer studyId) {
 		}
 	}
 
+	@Override
+	public List<StudyEntryDto> getStudyEntries(final Integer studyId, final StudyEntrySearchDto.Filter filter, final Pageable pageable) {
+		this.studyValidator.validate(studyId, false);
+		return this.studyGermplasmService.getStudyEntries(studyId, filter, pageable);
+	}
+
+	@Override
+	public long countAllStudyEntries(final Integer studyId) {
+		this.studyValidator.validate(studyId, false);
+		return this.studyGermplasmService.countStudyEntries(studyId);
+	}
+
+	@Override
+	public List<MeasurementVariable> getEntryDescriptorColumns(final Integer studyId) {
+		this.studyValidator.validate(studyId, false);
+		final Integer plotDatasetId = datasetService.getDatasets( studyId, new HashSet<>(Arrays.asList(DatasetTypeEnum.PLOT_DATA.getId()))).get(0).getDatasetId();
+
+		final List<MeasurementVariable> entryDescriptors =
+			this.datasetService.getObservationSetVariables(plotDatasetId, Lists
+				.newArrayList(VariableType.GERMPLASM_DESCRIPTOR.getId()));
+
+		//Remove OBS_UNIT_ID column and STOCKID if present
+		for (Iterator<MeasurementVariable> i = entryDescriptors.iterator(); i.hasNext();) {
```

Comment:
Keep terms to remove in a list then the for loop can be replaced by Collections.removeIf:
`entryDescriptors.removeIf(entry -> termsToRemove.contains(entry.getTermId()));`